src/components/SnippetArea.jsx

```diff
@@ -2,54 +2,61 @@ import React, { PropTypes } from 'react';
 import CodeMirror from 'react-codemirror';
 
 import { Card, CardText } from 'material-ui/Card';
+import '../styles/codesplain.css'
 import TextField from 'material-ui/TextField';
-
 import ConfirmLockDialog from './ConfirmLockDialog.jsx'
 import LockButton from './LockButton';
+import { getIndexToRowColConverter } from '../util/util.js';
 
 import 'codemirror/lib/codemirror.css';
-import 'codemirror/theme/material.css';
-
 import 'codemirror/mode/go/go.js';
 import 'codemirror/mode/python/python.js';
 
+
 const snippetEditorModes = {
   go: 'go',
   python3: 'python',
 };
 
-const SnippetArea = ({ contents, isDialogOpen, onTitleChanged, onSnippetChanged, readOnly, switchReadOnlyMode, snippetLanguage, toggleConfirmLockDialogVisibility }) => {
-  const codeMirrorOptions = {
-    lineNumbers: true,
-    theme: 'material',
-    mode: snippetEditorModes[snippetLanguage],
-    readOnly,
-  };
-  return (
-    <Card>
-      <CardText>
-      <TextField
-        name="snippetName"
-        hintText="Snippet Name"
-        onChange={onTitleChanged}
-      />
-      <LockButton
-        onClick={toggleConfirmLockDialogVisibility}
-        readOnly={readOnly}
-      />
-      <ConfirmLockDialog
-        isOpen={isDialogOpen}
-        accept={switchReadOnlyMode}
-        reject={toggleConfirmLockDialogVisibility}
-      />
-      <CodeMirror
-        value={contents}
-        options={codeMirrorOptions}
-        onChange={onSnippetChanged}
-      />
-      </CardText>
-    </Card>
-  );
+class SnippetArea extends React.Component {
+  constructor(props) {
+    super(props);
+    this.codeMirrorOptions = {
+      lineNumbers: true,
+      theme: 'codesplain',
+      mode: snippetEditorModes[this.props.snippetLanguage],
+      readOnly: this.props.readOnly,
+    };
+  }
+
+  render() {
+    return (
+      <Card>
+        <CardText>
+        <TextField
+          name="snippetName"
+          hintText="Snippet Name"
+          onChange={this.props.onTitleChanged}
+        />
+        <LockButton
+          onClick={this.props.toggleConfirmLockDialogVisibility}
+          readOnly={this.props.readOnly}
+        />
+        <ConfirmLockDialog
+          isOpen={this.props.isDialogOpen}
+          accept={this.props.switchReadOnlyMode}
+          reject={this.props.toggleConfirmLockDialogVisibility}
+        />
+        <CodeMirror
+          ref={cm => this.codeMirror = cm}
+          value={this.props.contents}
+          options={this.codeMirrorOptions}
+          onChange={ev => this.props.onSnippetChanged(ev, this.codeMirror)}
```

Comment:
I would prefer to move this to its own function on the class, so that we're not creating a new function each time we render.